storage/replica_command.go

```diff
@@ -325,21 +383,73 @@ func (r *Replica) EndTransaction(batch engine.Engine, ms *engine.MVCCStats, args
 			if ct.SplitTrigger != nil {
 				*ms = engine.MVCCStats{} // clear stats, as split will recompute from scratch.
 				if err := r.splitTrigger(batch, ct.SplitTrigger); err != nil {
-					return reply, err
+					return reply, nil, err
 				}
 			} else if ct.MergeTrigger != nil {
 				*ms = engine.MVCCStats{} // clear stats, as merge will recompute from scratch.
 				if err := r.mergeTrigger(batch, ct.MergeTrigger); err != nil {
-					return reply, err
+					return reply, nil, err
 				}
 			} else if ct.ChangeReplicasTrigger != nil {
 				if err := r.changeReplicasTrigger(ct.ChangeReplicasTrigger); err != nil {
-					return reply, err
+					return reply, nil, err
 				}
 			}
 		}
 	}
-	return reply, nil
+
+	return reply, externalIntents, nil
+}
+
+// intersectIntent takes an intent and a descriptor. It then splits the
```

Comment:
This feels premature. Why not just treat a range of intents which exceeds the bounds of the range as non-local and send it through the distributed sender, which already implements this kind of logic.